packages/cli/src/oclif/commands/promote.js

```diff
@@ -35,8 +45,24 @@ class PromoteCommand extends BaseCommand {
     const assumeYes = 'yes' in this.flags;
 
     let shouldContinue;
-    const changelog = await getVersionChangelog(version);
-    if (changelog) {
+    const { changelog, appMetadata, issueMetadata } = await getVersionChangelog(
+      version
+    );
+    if (!changelog) {
+      let error = `${colors.yellow(
+        'Warning!'
+      )} Changelog not found. Please create a CHANGELOG.md file in a format similar to ${colors.cyan(
+        'https://gist.github.com/xavdid/b9ede3565f1188ce339292acc29612b2'
+      )} with user-facing descriptions.`;
+      if (!appMetadata && !issueMetadata) {
```

Comment:
I'm not familiar with this feature, maybe good to have your team do a review as well.

Do we want to check for app/issue metadata only if changelog is falsey?

Reply:
Not sure what "check for" means, but the general logic is the following:

If we don't have a changelog, then we check if we don't have any metadata. If so that's appended to the outgoing warning.
If we do have a changelog, metadata isn't required alongside it as it's possible the change does not impact an action or specific issue.

Reply:
Sounds like metadata is optional, correct? If so, maybe the logic should be:

- If there's no changelog, print an error message
  - saying the user must provide a changelog
  - giving an example changelog that shows the metadata tag syntax
- If there's changelog, show the changelog and let the user confirm if they want to proceed to promote

Reply:
Sounds good to me